SHSearchBar/Classes/SHSearchBar.swift

```diff
@@ -332,11 +335,13 @@ extension SHSearchBar: UITextFieldDelegate {
     }
 
     public func textFieldShouldClear(_ textField: UITextField) -> Bool {
+        clearButtonDidClick?()
```

Comment:
Just below this line the delegate function is called to inform the outside world that the textfield has been cleared. Please use this function since it is tested.